grass7/raster/r.in.srtm.region/r.in.srtm.region.py

```diff
@@ -448,11 +536,28 @@ def main():
             grass.run_command('r.mapcalc', expression = '%s = round(%s)' % (output, output + '.float'))
             grass.run_command('g.remove', type = 'raster',
                               name = '%s,%s,%s' % (output + '.holes', output + '.interp', output + '.float'),
-                  flags = 'f',
+                              flags = 'f',
                               quiet = True)
 
     grass.run_command('g.remove', type = 'raster', pattern = pattern, flags = 'f', quiet = True)
 
+    # switch to target location
+    if kv['+proj'] != 'longlat':
+        os.environ['GISRC'] = str(TGTGISRC)
+        # r.proj
+        grass.message(_("Reprojecting <%s>...") % output)
+        res = 30 if srtmv3 else 90
```

Comment:
30 or 90 is a bad choice because this assumes meters as units, other possible units are
meters:meter:1.0
feet:foot:0.3048
miles:mile:1609.344
inches:inch:2.540000e-02
centimeters:centimeter:0.01
nanometers:nanometer:1.000000e-09
microns:micron:1.000000e-06
angstroms:angstrom:1.000000e-10
decinanometers:decinanometer:1.000000e-10
yards:yard:0.9144
rods:rod:5.0292
lightyears:lightyear:9.460530e+15
USfeet:USfoot:0.30480060960121920243
USinches:USinch:.02540005080010160020
USyards:USyard:.91440182880365760731
USmiles:USmile:1609.34721869443738887477

also with some duplicates
    {"km",      "1000.",                "Kilometer",                    1000.0},
    {"m",       "1.",                   "Meter",                        1.0},
    {"dm",      "1/10",                 "Decimeter",                    0.1},
    {"cm",      "1/100",                "Centimeter",                   0.01},
    {"mm",      "1/1000",               "Millimeter",                   0.001},
    {"kmi",     "1852.0",               "International Nautical Mile",  1852.0},
    {"in",      "0.0254",               "International Inch",           0.0254},
    {"ft",      "0.3048",               "International Foot",           0.3048},
    {"yd",      "0.9144",               "International Yard",           0.9144},
    {"mi",      "1609.344",             "International Statute Mile",   1609.344},
    {"fath",    "1.8288",               "International Fathom",         1.8288},
    {"ch",      "20.1168",              "International Chain",          20.1168},
    {"link",    "0.201168",             "International Link",           0.201168},
    {"us-in",   "1./39.37",             "U.S. Surveyor's Inch",         0.0254},
    {"us-ft",   "0.304800609601219",    "U.S. Surveyor's Foot",         0.304800609601219},
    {"us-yd",   "0.914401828803658",    "U.S. Surveyor's Yard",         0.914401828803658},
    {"us-ch",   "20.11684023368047",    "U.S. Surveyor's Chain",        20.11684023368047},
    {"us-mi",   "1609.347218694437",    "U.S. Surveyor's Statute Mile", 1609.347218694437},
    {"ind-yd",  "0.91439523",           "Indian Yard",                  0.91439523},
    {"ind-ft",  "0.30479841",           "Indian Foot",                  0.30479841},
    {"ind-ch",  "20.11669506",          "Indian Chain",                 20.11669506},

1 arcsecond is appr. 30 meters only at the equator but not e.g. in central Europe where it is again different to northern Europe. The user must decide on an appropriate target resolution for reprojection. Related: do not make bilinear_f the default resampling method, only the user can know which is the best method considering the change in resolution and use case.

Reply:
yes, agreed